### DISCUSSION TITLE
Is there an example of OIDC auth without using Auth0?

### DISCUSSION THREAD

**Discussion by jaredlll08** [vercel/next.js#11820] [Help]
Hi,

I am trying to implement OpenID Connect on my website but have had some trouble with it. All the documentation and references I can find use Auth0 however I am using IdentityServer4 as my identity provider. I would prefer to avoid using Auth0 where possible although I don't mind using their open source libraries. 

The main problem is that I can only find examples using Auth0's identity provider service. I've tried cloning a few of the example next+auth0 repos and configuring the client to use IS4. This works for the most part however the Auth0 client calls the `/v2/logout` endpoint which is not available with IS4. 

Any help would be greatly appreciated.

**Comment by empz** [comment]
I'm also interested in this. We really need a generic example using any of the node/browser OIDC libraries.

**Comment by balazsorban44** [comment]
https://github.com/nextauthjs/next-auth Could be a good solution, it has built-in support for IdentityServer4 (among many others)

(We use both of these together at work, and it's very nice)

You can also set `idToken: true` in the [provider options](https://next-auth.js.org/configuration/providers#oauth-provider-options), and if you enable always sending claims in the `id_token`, `next-auth` will parse those for you.

**Comment by realattila** [comment]
i want to use IdentityServer4 cliend mod not server mod, i have other backend that check user with token generated by user, i do not want use secret key, want to redirect to login page SSO, can i use next-auth ?

**Reply by balazsorban44** [reply]
yes
